### PR TITLE
Retry: Student view only push

### PR DESCRIPTION
**Pull request by shamedone**
Hi Dr Puder, 

Ok so I looked over what you wrote and am retrying the student view only pull request again. I have squashed all the related commits together and this pull only looks at provide the resources for a student read_only_view, although the button to access it is still commented out in this pull as it does not serve a real a purpose atm. 

I just want to be clear on best practices, I have sorta restarted and am doing things in this process for each feature I am working on: create feature branch, edit and commit as much as I wish until the feature is done. Then merge back into my master branch and create a pull request. If that works, then I will follow that model as I continue to update the page. 

As always, let me know if you have any questions. If this process looks good I will begin sending more pull requests in smaller feature size chunks. Thanks!

 -Tom

### CODE REVIEW

File: src/main/java/edu/sfsu/AdvisingServlet.java [hunk 1]
```diff
@@ -71,16 +71,29 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
             return;
         }
 
+        if (path.equals("/lookup-student")) {
+            String id = request.getParameter("id");
+            String token = request.getParameter("token");
+            String readonly = request.getParameter("readonly");
```
Comment (apuder) [inline src/main/java/edu/sfsu/AdvisingServlet.java:77]: It is not a good idea to let the client tell us whether the view should be read-only or not. That can easily be hacked by a tech-savvy person. The server should "know" whether the currently logged-in user is a superuser (read/write access) or a student (read-only access). Ultimately I want to integrate this with the campus SSO. But for now I suggest to "fake" something. Perhaps add a second password for read-only access in the property file.

File: src/main/java/edu/sfsu/HtmlFormatter.java [hunk 2]
```diff
@@ -110,6 +113,32 @@ public static String generateHtml(Student student) {
 
         fragment.p("comments", student.comment);
 
+        generateCheckpoint(fragment, student.id, student.checkpointOralPresentation,
```
Comment (apuder) [inline src/main/java/edu/sfsu/HtmlFormatter.java:116]: Why do you need to duplicate these lines? A little further below there is exactly the same statement.

File: src/main/webapp/index.html [hunk 3]
```diff
@@ -82,19 +90,27 @@
             printWindow.document.close();
         }
 
-        function lookup_student() {
+        function lookup_student(readonly) {
             $('#student_id').blur();
             var id = document.getElementById("student_id");
             var section = document.getElementById("section_student_info");
             var info = document.getElementById("student_info");
+            if (readonly =="true"){
```
Comment (apuder) [inline src/main/webapp/index.html:98]: Can you please do consistent formatting?  Please insert spaces where appropriate. Also do this for Java code.

File: src/main/webapp/index.html [hunk 4]
```diff
@@ -278,14 +298,32 @@ <h4>413 Advising List</h4>
...
-                      allow to specify a date range.</p>
+                        allow to specify a date range.</p>
                     <button class="mdl-button mdl-js-button mdl-button--fab" onclick="return generateList(this, 'graduated')">
                         <i class="material-icons">forward</i>
                     </button>
                 </div>
             </section>
         </div>
     </main>
+    <div class="mdl-layout__tab-panel" id="senior_advising_read_only" style="display: none">
```
Comment (apuder) [inline src/main/webapp/index.html:309]: I suggest you remove all this HTML and decide on the server side (based on which password the user typed in) whether you send the editable or read-only HTML. I don't see the need for duplicating this HTML.

**Review by apuder — CHANGES_REQUESTED** [review]
Sorry for the slow feedback. I left come comments. Please make the necessary changes. In general: one PR should focus on one feature and should contain only one commit. How you organize your branches and commits at your end is up to you. The log message for the one commit should give a high-level summary. No need to document the individual steps.

**Comment by shamedone** [comment]
Hi Dr Puder, 

Thanks for your feedback. As I looked over this code and feedback, it strikes me that this read only view feature was more of a way for me to get my to get familiar with the software than a change that was actually requested from you or Dr. Hsu. So what I suggest is maybe closing or setting aside this pull request for the time being, and then me sending a new pull request for one of the other features I have worked on, that way you can look at some of the more useful changes while I can reevaluate how best to implement the comments you outlined in the above. Let me know! Thanks!

 -Tom

**Comment by apuder** [comment]
OK, I will close this PR for now. Perhaps we should first discuss the scope of each PR. E.g., I suggest to make the test/dummy DB its own PR. This way it will be much easier for me to understand and follow your changes.

**Pull request closed without merging**